lib/rules/jest-enzyme-matchers.js

```diff
@@ -0,0 +1,232 @@
+const t = require("@babel/types");
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "Requires the use more of enzyme matchers where appropriate.",
+            category: "jes",
+            recommended: false,
+        },
+        fixable: true,
+    },
+
+    create(context) {
+        const stack = [];
+
+        return {
+            CallExpression(node) {
+                if (
+                    t.isIdentifier(node.callee, {name: "expect"}) &&
+                    t.isCallExpression(node.arguments[0]) &&
+                    t.isMemberExpression(node.arguments[0].callee) &&
+                    t.isMemberExpression(node.parent) &&
+                    t.isIdentifier(node.parent.property) &&
+                    t.isCallExpression(node.parent.parent)
+                ) {
+                    const {property} = node.arguments[0].callee;
+                    const matcher = node.parent.property.name;
+
+                    if (matcher === "toEqual") {
+                        if (t.isIdentifier(property, {name: "prop"})) {
```

Comment:
Wow, having to handle each type individually is a lot!

Reply:
If I add more cases in the future I'll try to dedupe things a bit.